// @usedapp/core/dist/cjs/src/model/chain/aurora.js
"use strict";
exports.__esModule = true;
exports.AuroraTestnet = exports.Aurora = void 0;
exports.Aurora = {
    chainId: 1313161554,
    chainName: 'Aurora',
    isTestChain: false,
    isLocalChain: false,
    multicallAddress: '0x32b50c286DEFd2932a0247b8bb940b78c063F16c',
    multicall2Address: '0xace58a26b8Db90498eF0330fDC9C2655db0C45E2',
    getExplorerAddressLink: function (address) { return "https://explorer.mainnet.aurora.dev/address/".concat(address); },
    getExplorerTransactionLink: function (transactionHash) { return "https://explorer.mainnet.aurora.dev/tx/".concat(transactionHash); }
};
exports.AuroraTestnet = {
    chainId: 1313161555,
    chainName: 'Aurora Testnet',
    isTestChain: true,
    isLocalChain: false,
    multicallAddress: '0x755E730F28A31079711aB588e3568e70A40F3564',
    getExplorerAddressLink: function (address) { return "https://explorer.testnet.aurora.dev/address/".concat(address); },
    getExplorerTransactionLink: function (transactionHash) { return "https://explorer.testnet.aurora.dev/tx/".concat(transactionHash); }
};
exports["default"] = {
    Aurora: exports.Aurora,
    AuroraTestnet: exports.AuroraTestnet
};
