// @usedapp/core/dist/cjs/src/model/chain/avalanche.js
"use strict";
exports.__esModule = true;
exports.AvalancheTestnet = exports.Avalanche = void 0;
exports.Avalanche = {
    chainId: 43114,
    chainName: 'Avalanche',
    isTestChain: false,
    isLocalChain: false,
    multicallAddress: '0xdDCbf776dF3dE60163066A5ddDF2277cB445E0F3',
    getExplorerAddressLink: function (address) { return "https://snowtrace.io/address/".concat(address); },
    getExplorerTransactionLink: function (transactionHash) { return "https://snowtrace.io/tx/".concat(transactionHash); }
};
exports.AvalancheTestnet = {
    chainId: 43113,
    chainName: 'AvalancheTestnet',
    isTestChain: true,
    isLocalChain: false,
    multicallAddress: '0xccc75e78Dce6A20bCCa3a30deB23Cb4D23df993a',
    getExplorerAddressLink: function (address) { return "https://testnet.snowtrace.io/address/".concat(address); },
    getExplorerTransactionLink: function (transactionHash) { return "https://testnet.snowtrace.io/tx/".concat(transactionHash); }
};
exports["default"] = {
    Avalanche: exports.Avalanche,
    AvalancheTestnet: exports.AvalancheTestnet
};
